### PR TITLE
Finish task callback record integration

### PR DESCRIPTION
**Pull request by tomyhoi**
- Remove last part of task tracking code I added
- Extended some "events" to include more data from the files
- Finished integrating new task events into processing (DataModel)
- Improved open code

### CODE REVIEW

**Comment by tomyhoi** [comment]
@gbtitus short changes, review if you like

File: tools/chplvis/DataModel.cxx [hunk 1]
```diff
@@ -316,23 +336,72 @@ int DataModel::LoadData(const char * filename)
...
       case Ev_task:
         tp = (E_task *)ev;
-        if (++(curTag->locales[curNodeId].numTasks) > curTag->maxTasks)
-          curTag->maxTasks = curTag->locales[curNodeId].numTasks;
-        if (++(tagList[0]->locales[curNodeId].numTasks) > tagList[0]->maxTasks)
-          tagList[0]->maxTasks = tagList[0]->locales[curNodeId].numTasks;
+        //        if (++(curTag->locales[curNodeId].numTasks) > curTag->maxTasks)
+        //          curTag->maxTasks = curTag->locales[curNodeId].numTasks;
+        //        if (++(tagList[0]->locales[curNodeId].numTasks) > tagList[0]->maxTasks)
+        //          tagList[0]->maxTasks = tagList[0]->locales[curNodeId].numTasks;
```
Comment (gbtitus) [inline tools/chplvis/DataModel.cxx]: Did you really want to keep these lines for some reason (future use), or were you intending to remove them?

**Comment by gbtitus** [comment]
This looks good, though I definitely reviewed more carefully in the runtime part than the tools part. If you have someone else who has been reviewing the tools part it would be good for them to take a look also.

File: tools/chplvis/DataModel.cxx (hunk 1 above)
Comment (tomyhoi) [inline tools/chplvis/DataModel.cxx]: Good point.   Thanks.

**Pull request merged**